ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingPropertySource.java

```diff
@@ -206,64 +203,42 @@ public void initialize() {
 		}
 	}
 
-	public TracingPropertySource(IPluginModelBase model, Properties masterOptions, Hashtable<String, String> template,
-			TracingBlock block) {
+	public TracingPropertySource(IPluginModelBase model, Map<String, String> masterOptions,
+			Map<String, String> template, TracingBlock block) {
 		fModel = model;
 		fMasterOptions = masterOptions;
 		fTemplate = template;
 		fBlock = block;
-		fValues = new Hashtable<>();
+		fValues = new HashMap<>();
 	}
 
 	public IPluginModelBase getModel() {
 		return fModel;
 	}
 
-	private Object[] getSortedKeys(int size) {
-		Object[] keyArray = new Object[size];
-		int i = 0;
-		for (Enumeration<String> keys = fTemplate.keys(); keys.hasMoreElements();) {
-			String key = keys.nextElement();
-			keyArray[i++] = key;
-		}
-		Arrays.sort(keyArray, this::compareKeys);
-		return keyArray;
-	}
-
-	private int compareKeys(Object o1, Object o2) {
-		String s1 = (String) o1;
-		String s2 = (String) o2;
-		// equal
-		return s1.compareTo(s2);
-	}
-
 	public void createContents(Composite parent, boolean enabled) {
-		fDescriptors = new Vector<>();
+		List<PropertyEditor> fDescriptors = new ArrayList<>();
 		TableWrapLayout layout = new TableWrapLayout();
 		layout.numColumns = 2;
 		layout.rightMargin = 10;
 		layout.leftMargin = 10;
 		parent.setLayout(layout);
 		boolean bordersNeeded = false;
-		Object[] sortedKeys = getSortedKeys(fTemplate.size());
-		for (Object keyObject : sortedKeys) {
-			String key = (String) keyObject;
-			IPath path = IPath.fromOSString(key);
-			path = path.removeFirstSegments(1);
-			String shortKey = path.toString();
+		Iterable<String> sortedKeys = fTemplate.keySet().stream().sorted()::iterator;
+		for (String key : sortedKeys) {
+			String shortKey = IPath.fromOSString(key).removeFirstSegments(1).toString();
 			String value = fTemplate.get(key);
 			String lvalue = null;
-			String masterValue = fMasterOptions.getProperty(key);
-			String commentValue = fMasterOptions.getProperty("#" + key); //$NON-NLS-1$
+			String masterValue = fMasterOptions.get(key);
+			String commentValue = fMasterOptions.get("#" + key); //$NON-NLS-1$
 			PropertyEditor editor;
-			if (value != null)
+			if (value != null) {
 				lvalue = value.toLowerCase(Locale.ENGLISH);
+			}
 			if (lvalue != null && (lvalue.equals("true") || lvalue.equals("false"))) { //$NON-NLS-1$ //$NON-NLS-2$
 				editor = new BooleanEditor(shortKey, shortKey, commentValue);
 				if (masterValue != null) {
-					Integer mvalue = Integer.valueOf(masterValue.equals("true") //$NON-NLS-1$
-					? 1
-							: 0);
+					Integer mvalue = Boolean.parseBoolean(masterValue) ? 1 : 0;
```

Comment:
FYI `Boolean.parseBoolean` uses `equalsIgnoreCase` so this change is **not** a refactoring and this line is now **not** in sync with the rest of the the class (_e.g._ with the ones in line 238). 

I would either revert this change and keep it consistent with the rest of the logic in the class (**case sensitive** _i.e._ "true" is `true` but "TrUe" is `false`) or I would adapt the logic in the other places and let the whole class work in a **case insensitive** manner (_i.e._ "true" is `true` and "TrUe" is also `true`). 

Your call :-)

Reply:
Thanks for this remark, that's a good point as well.
But looking at the vicinity of this line, especially line 238 it contains `lvalue.equals("true") || lvalue.equals("false")` and `lvalue` is `lvalue = value.toLowerCase(Locale.ENGLISH)`, so at least that is already comparing not case-sensitive, just in a more complicated way (btw. using toLowerCase seems to be the pattern used before `equalsIgnoreCase()` case existed). But using `equalsIgnoreCase` allows to remove `lvalue` as well and operate on `value` directly.

In general I wonder why boolean values are stored as Integer values in the fValues Map<String,Object> while it could simply use Booleans?
Just using Booleans would also simplify the save() method below because the Boolean could be just converted into a String and there would be no need to convert the int to a `true` or `false` String again.

Reply:
> But looking at the vicinity of this line, especially line 238...

👍 

> In general I wonder why boolean values are stored as Integer values in the fValues Map<String,Object> while it could simply use Booleans?

`0`s and `1`s were probably allowed in the configuration file before and since the method `getOptions(IPluginModelBase)` doesn't convert them to "real" `boolean`s then the logic had to be moved someplace else. If that's the case then removing the logic should be considered a breaking change.

Reply:
On the plus side: I don't see anywhere in the documentation that 0s and 1s are allowed to be used as booleans and I don't see anywhere that one writes 0s and 1s into the file so it should be fine to delete it.

Or do you see any place were 0s and 1s are written into the file?

Reply:
> Or do you see any place were 0s and 1s are written into the file?

Nope. And from the code I don't think it is even possible, so this change seems save.